packages/agreement-process/test/testUpgradeAgreement.ts

```diff
@@ -0,0 +1,1339 @@
+      }
+
+      expect(actualAgreementData).toMatchObject({
+        ...expectedStoreEvent,
+        event_version: 1,
+      });
+
+      return decoderFunction(actualAgreementData, expectedStoreEvent);
+    }
+
+    function toAgreementStampsV1(agreement: Agreement): StampsV1 {
+      const stamps: StampsV1 = {};
+
+      if (agreement.stamps.submission) {
+        stamps.submission = toStampV1(agreement.stamps.submission);
+      }
+
+      if (agreement.stamps.activation) {
+        stamps.activation = toStampV1(agreement.stamps.activation);
+      }
+
+      if (agreement.stamps.rejection) {
+        stamps.rejection = toStampV1(agreement.stamps.rejection);
+      }
+
+      if (agreement.stamps.suspensionByProducer) {
+        stamps.suspensionByProducer = toStampV1(
+          agreement.stamps.suspensionByProducer
+        );
+      }
+
+      if (agreement.stamps.suspensionByConsumer) {
+        stamps.suspensionByConsumer = toStampV1(
+          agreement.stamps.suspensionByConsumer
+        );
+      }
+
+      if (agreement.stamps.upgrade) {
+        stamps.upgrade = toStampV1(agreement.stamps.upgrade);
+      }
+
+      if (agreement.stamps.archiving) {
+        stamps.archiving = toStampV1(agreement.stamps.archiving);
+      }
+
+      return stamps;
+    }
+
+    async function uploadDocument(path: string, name: string): Promise<void> {
+      await fileManager.storeBytes(
+        config.s3Bucket,
+        config.consumerDocumentsPath,
+        path,
+        name,
+        Buffer.from("large-document-file")
+      );
+      expect(await fileManager.listFiles(config.s3Bucket)).toContainEqual(
+        `${config.consumerDocumentsPath}/${path}/${name}`
+      );
+    }
+
+    it("should succeed with valid Verified and Declared attributes", async () => {
+      const authData = {
+        ...getRandomAuthData(),
+        userRoles: [userRoles.ADMIN_ROLE],
+      };
+      const tenantId = authData.organizationId.toString();
+
+      const validVerifiedTenantAttribute = {
+        ...getMockVerifiedTenantAttribute(unsafeBrandId<AttributeId>(tenantId)),
+        verifiedBy: [
+          {
+            id: unsafeBrandId<TenantId>(tenantId),
+            verificationDate: new Date(TEST_EXECUTION_DATE.getFullYear() - 1),
+            expirationDate: new Date(TEST_EXECUTION_DATE.getFullYear() + 1),
+            extensionDate: undefined,
+          },
+        ],
+      };
+      const validVerifiedEserviceAttribute = getMockEServiceAttribute(
+        validVerifiedTenantAttribute.id
+      );
+      const validDeclaredTenantAttribute = {
+        ...getMockDeclaredTenantAttribute(unsafeBrandId<AttributeId>(tenantId)),
+        revocationTimestamp: undefined,
+      };
+      const validDeclaredEserviceAttribute = getMockEServiceAttribute(
+        validDeclaredTenantAttribute.id
+      );
+
+      const validCertifiedTenantAttribute = getMockCertifiedTenantAttribute(
+        unsafeBrandId<AttributeId>(tenantId)
+      );
```

Comment:
This can generate invalid attributes because you are not setting the `revocationTimestamp` to undefined, I am not sure why this test is passing 🤔 Same in other parts of the app where you create valid certified attributes (see also https://github.com/pagopa/interop-be-monorepo/pull/384/files#diff-eb1c55d89587fb4c44d3cdd3aba496928591f7faf95594b8e9cfd422f6eaafb5R65 as reference)